independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartSpec.java

```diff
@@ -15,22 +15,28 @@
 public final class CodestartSpec {
 
     enum Type {
-        PROJECT(true),
-        LANGUAGE(true),
-        BUILDTOOL(true),
-        CONFIG(true),
-        CODE(false),
-        TOOLING(false);
+        PROJECT(true, 3),
+        LANGUAGE(true, 1),
+        BUILDTOOL(true, 2),
+        CONFIG(true, 4),
+        CODE(false, 6),
+        TOOLING(false, 5);
```

Comment:
Dumb question, but isn't easier to just use `Enum.ordinal()` and reorder the items?

Reply:
@gastaldi, I guess that would aso do, but this makes it clear that we actually use the order if someone adds/change a value IMO

Reply:
IMHO it is more likely someone could introduce the same `order` number than change the enum order (which is more visible) :rofl: 

But it's your call :wink:

Reply:
@gastaldi introducing the same would be alright, it means they are the same level and would be compared by name

Reply:
@gastaldi I think if no one else objects I will stick to that implementation. beside that, is the PR ok? could you approve the PR?

Reply:
I agree having an explicit order is better. I could see how someone could reorder them alphabetically without noticing.

Now the question is: why are they in a random order? Can't we properly order them too?

Reply:
My question is not only about it being nicer, having things in a random order makes adding new items more error prone.

Reply:
@gsmet the given order is not random:
- language is first as it defines the language for the others
- buildtool is second as it is also important for the others
- project then can give overrides on those
- config and tooling could be at the same level
- code needs to be last

For now I think it doesn't really matter as soon as the order is consistent between run, but I am creating an issue for this as the order for data overriding may need to be different from the order of processing. I don't think we want the examples codestarts to override base data.

Reply:
What I'm saying is that the order of the enum is random.

The enum should be sorted by execution order so that it's less error prone.

Reply:
ah ok :)

Reply:
I created an issue for the data overriding order as it may also be important: https://github.com/quarkusio/quarkus/issues/11754